File: 392-is-subsequence/is-subsequence.c
bool isSubsequence(char* s, char* t) {
    int j = 0;
    int i = 0;
    int s_len = strlen(s);
    int t_len = strlen(t);
    while (i < s_len && j < t_len) {
        if (s[i] == t[j]) {
            i++;
            j++;
        } else {
            j++;
        }
    }
    return i == s_len;
}

File: 392-is-subsequence/is-subsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        int S = 0;
        int T = 0;

        while(S<s.length() && T<t.length()){
            if(s.charAt(S) == t.charAt(T)){
                S++;
            } T++;

        }
        return S == s.length();
    }
}